### ISSUE TITLE
node.duplicate() function triggers set_name == "" error

### ISSUE THREAD

**Issue by burstina** [godotengine/godot#32449]
**Godot version:**
3.1.1

**OS/device including version:**
Win10

**Issue description:**
Using duplicate() function over a not named node triggers console message
```
ERROR: set_name: Condition ' name == "" ' is true.
   At: scene/main/node.cpp:930
```
but it is not an error: the newly created node still get unique name and project execution continues flawlessly.

adding ```nod.set_name("something") ``` before duplicate() allows to bypass the error message

**Steps to reproduce:**
mrp provided
run the project and look at console and remote tree

**Minimal reproduction project:**
[MRP_duplicate_set_name.zip](https://github.com/godotengine/godot/files/3672670/MRP_duplicate_set_name.zip)

**Comment by Anutrix** [comment]
@burstina Unrelated but a request. Please do not zip a file. It doesn't get compressed further. Also, it causes issues with Windows Explorer's Archive Manager.

**Comment by Anutrix** [comment]
Couldn't reproduce in latest master https://github.com/godotengine/godot/commit/79298face0d7dc53b3c014ac910212a5993c1d1c. It's most probably fixed.

**Comment by akien-mga** [comment]
Indeed, duplicate of #27319, fixed by #29246. Thanks for testing and looking it up.

**Comment by Anutrix** [comment]
Happy to help.
